examples/tasks/async_compute.rs

```diff
@@ -0,0 +1,122 @@
+use bevy::prelude::*;
+use bevy::tasks::{AsyncComputeTaskPool, Task};
+use std::time::{Instant, Duration};
+use rand::Rng;
+use std::option::Option::Some;
+use futures_lite::future;
+
+/// This example shows how to use the ecs system and the AsyncComputeTaskPool
+/// to spawn, poll, and complete tasks across systems and system ticks.
+
+// Number of cubes to spawn across the x, y, and z axis
+const NUM_CUBES: i32 = 6;
+
+// Used to tag our new entity spawned with tasks
+struct Marker;
+
+/// This system generates tasks simulating computationally intensive
+/// work that potentially spawns multiple frames/ticks. A separate
+/// system, handle_tasks, will poll the spawned tasks on subsequent
+/// frames/ticks, and use the results to spawn cubes
+fn spawn_tasks(
+    mut commands: Commands,
+    thread_pool: Res<AsyncComputeTaskPool>,
+) {
+    for x in 0 ..NUM_CUBES {
+        for y in 0 ..NUM_CUBES {
+            for z in 0 ..NUM_CUBES {
+
+                // Spawn new task on the AsyncComputeTaskPool
+                let task = thread_pool.spawn(async move {
+
+                    let mut rng = rand::thread_rng();
+                    let start_time = Instant::now();
+                    let duration = Duration::from_secs_f32(rng.gen_range(0.05..0.2));

```

Comment:
This could use a comment explaining that we're sleeping to demonstrate the functionality :)

Reply:
Added in 846b63b

Reply:
@alice-i-cecile Is the new commentary ok?

Reply:
Yep looks good to me.